### ISSUE TITLE
Difficult to trace why release 1.0.4 broke my app with message: ERROR: $ operator is invalid for atomic vectors

### ISSUE THREAD

**Issue by jgbaum** [rstudio/shiny#1841]
Hi,

I've been working on a shiny app (https://github.com/LJI-Bioinformatics/Shiny-PCA-Maker) that runs just fine under shiny 1.0.1 through 1.0.3.  However, upon upgrading to 1.0.4 it will not run.  Instead, this appears in the browser:

`ERROR: $ operator is invalid for atomic vectors`

And here is what is in the console:

> 
> Listening on http://127.0.0.1:6889
> Warning: Error in $: $ operator is invalid for atomic vectors
> Stack trace (innermost first):
>     57: tag
>     56: tags$a
>     55: tag
>     54: tags$li
>     53: FUN
>     52: lapply
>     51: buildTabset
>     50: tabsetPanel
>     49: tag
>     48: tags$div
>     47: div
>     46: tagList
>     45: attachDependencies
>     44: bootstrapPage
>     43: fluidPage
>      1: runApp
> Error : $ operator is invalid for atomic vectors
> Warning: Error in $: $ operator is invalid for atomic vectors
> Stack trace (innermost first):
>     57: tag
>     56: tags$a
>     55: tag
>     54: tags$li
>     53: FUN
>     52: lapply
>     51: buildTabset
>     50: tabsetPanel
>     49: tag
>     48: tags$div
>     47: div
>     46: tagList
>     45: attachDependencies
>     44: bootstrapPage
>     43: fluidPage
>      1: runApp
> Error : $ operator is invalid for atomic vectors

I've systematically gone through the shiny releases from 1.0.1 through 1.0.5 and the code runs fine up to 1.0.3.  This behavior is only seen on 1.0.4 and 1.0.5.  Any ideas on how to solve this issue would be much appreciated.

Thanks,

J

**Comment by jgbaum** [comment]
I've figured it out!  In my tabPanel call, I was passing an empty string as one of the elements.

> tabsetPanel("", ...)

Previous versions of shiny did not complain about this or throw any errors.  It seems additional validation must have been added for correct formatting of tabsetPanel elements.  It would be nice if there were a meaningful error message for this issue, but I understand how that might be beyond the scope of the framework.  In any case, I would consider this issue closed.

Thanks!

-J
